database/rrdset.c

```diff
@@ -1332,8 +1339,11 @@ static inline size_t rrdset_done_interpolate(
                     break;
             }
 
+            time_t curr_t = (time_t)(next_store_ut / USEC_PER_SEC);
```

Comment:
This is a small one, only to think. We are using `time_t` and we created `ml_host_t `, so we are using the suffix `_t` to define types, so I suggest to sue a different name for `curr_t` avoiding confusion between data type and variable names.